core/src/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -145,6 +145,20 @@ public Object eval(DataInstance model, EvaluationContext evalContext) {
 
         Hashtable funcHandlers = evalContext.getFunctionHandlers();
 
+        for (int i = 0; i < args.length; i++) {
+            argVals[i] = args[i].eval(model, evalContext);
+        }

```

Comment:
You shouldn't move argument evaluation in above of if/else expression short-circuiting that occurs on line 162,